### PR TITLE
TKWaypointRouter now send shared vehicle ID

### PR DESCRIPTION
**Pull request by nighthawk**
Refactor and review TKWaypointRouter, adding explicit Swift model for data input rather than using `[String: Any]`.

Backend API documentation update is here: https://github.com/skedgo/tripgo-api/pull/224

### CODE REVIEW

File: Sources/TripKit/server/TKWaypointRouter.swift [hunk 1]
```diff
@@ -344,26 +306,131 @@ extension TKWaypointRouter {
...
 
-class WaypointParasBuilder {
+// MARK: - Input
+
+extension TKWaypointRouter {
+  
+  struct Input: Encodable {
+    var segments: [Segment]
+    var vehicles: [TKAPI.PrivateVehicle]
+    // var config: TKConfig
```
Comment (nighthawk) [inline Sources/TripKit/server/TKWaypointRouter.swift:316]: This can be used when we merged this into `main`. I'll create a separate PR for that.

Comment (julesian) [reply Sources/TripKit/server/TKWaypointRouter.swift:316]: noted

**Comment by julesian** [comment]
does `paras` mean parameters?

looks good to me

**Review by julesian — APPROVED** [review]

**Comment by nighthawk** [comment]
> does `paras` mean parameters?

Yep!

**Pull request merged**